File: ejercicios hechos/Python/ejercicios repeticiones/act6.py
#Programa: act6.py
#Propósito: Escribe un programa que dados dos números, uno real (base) y un entero positivo (exponente), saque por pantalla el resultado de la potencia. No se puede utilizar el operador de potencia.
#Autor: Jose Manuel Serrano Palomo.
#Fecha: 29/10/2019
#
#Variables a usar:
# b es la base, p es la potencia, exp el exponente, i es el indice del bucle 
#
#Algoritmo:
# Leemos b y p
# Nos aseguramos de que él exoponente es positivo
# calculamos la potencia acumulando el productor de la base en cuestión del exponente

print("Realizar una potencia sin el operador potencia")
print("-----------------------------------------------\n")

#Leemos datos
b = int(input("Dime la base de la potencia: "))
exp = int(input("Dime el exponente de la potencia: "))

while exp < 0:
    exp = int(input("El exponente no puede ser negativo: "))
    
p = 1

for i in range(0, exp):
    p = p * b
    
print(f"El resultado es {p}")

File: ejercicios terminados/Java/estructuras repetitivas/act6.java
/*Programa: act6.java
 * 
 *Propósito: Escribe un programa que dados dos números, uno real (base) y 
 * un entero positivo (exponente), saque por pantalla el resultado de la 
 * potencia. No se puede utilizar el operador de potencia.
 * 
 *Autor: Jose Manuel Serrano Palomo.
 * 
 *Fecha: 10/11/2019
 *
 *Análisis:
 * Leemos la base y exponente de dos numeros y debemos realizar la potencia
 *
 *Diseño:
 * 1. Inicializo el acumulador para multiplicación -> potencia <- 1
 * 2. Leer base
  * 3. Repetir -> Leer exponente -> hasta que exponente sea positivo
  * 4. Desde 1 hasta el exponente
  * 5. Acumulo la multiplicación de la base -> potencia<-potencia*base
  * 6. Muestro potencia
*/
import java.util.Scanner;
public class act6 {
    public static void main(String[] args) {
        System.out.println("Programa que calcule la potencia");
        System.out.println("--------------------------------");
        Scanner s = new Scanner(System.in);
        double base;
        int exponente;
        double potencia;

        System.out.print("Dame la base de la potencia: ");
        base = s.nextDouble();
        do {
          System.out.print("Dame el exponente de la potencia: ");
          exponente = s.nextInt();
          if (exponente<0) {
            System.out.println("ERROR: El exponente debe ser positivo");
          }
        } while (exponente<0);

        potencia = 1;
        for (int i=1;i<=exponente;i++) {
          potencia *= base;
        }
        System.out.println("Potencia: "+potencia);
      }
    }
}
